allennlp/common/sqlite_sparse_sequence.py

```diff
@@ -0,0 +1,102 @@
+import os
+import shutil
+from os import PathLike
+from typing import MutableSequence, Any, Union, Iterable
+
+from sqlitedict import SqliteDict
+
+
+class SqliteSparseSequence(MutableSequence[Any]):
+    def __init__(self, filename: Union[str, PathLike], read_only: bool = False):
+        self.table = SqliteDict(filename, "sparse_sequence", flag="r" if read_only else "c")
+
+    def __del__(self):
+        self.close()
+
+    def __getitem__(self, i: Union[int, slice]) -> Any:
+        if isinstance(i, int):
+            try:
+                return self.table[str(i)]
+            except KeyError:
+                current_length = len(self)
+                if i >= current_length or current_length <= 0:
+                    raise IndexError("list index out of range")
+                elif i < 0 < current_length:
+                    return self.__getitem__(i % current_length)
+                else:
+                    return None
+        elif isinstance(i, slice):
+            from allennlp.tango.dataloader import ShuffledSequence
```

Comment:
Is it necessary to put this import here? I'm afraid this would add non-negligible runtime overhead each time this path is reached.

Reply:
I made it a global import.